rcl/src/rcl/arguments.c

```diff
@@ -354,6 +354,45 @@ rcl_get_zero_initialized_arguments(void)
   return default_arguments;
 }
 
+rcl_ret_t
+rcl_remove_ros_arguments(
+  char const * const argv[],
+  rcl_arguments_t * args,
+  rcl_allocator_t allocator,
+  int * nonros_argc,
+  const char ** nonros_argv[])
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(argv, RCL_RET_INVALID_ARGUMENT, allocator);
+  RCL_CHECK_ARGUMENT_FOR_NULL(nonros_argc, RCL_RET_INVALID_ARGUMENT, allocator);
+  RCL_CHECK_ARGUMENT_FOR_NULL(args, RCL_RET_INVALID_ARGUMENT, allocator);
+  RCL_CHECK_ALLOCATOR_WITH_MSG(&allocator, "invalid allocator", return RCL_RET_INVALID_ARGUMENT);
+
+  *nonros_argc = rcl_arguments_get_count_unparsed(args);
+  if (*nonros_argc > 0) {
+    int * unparsed_indices = NULL;
+    rcl_ret_t ret;
+    ret = rcl_arguments_get_unparsed(args, allocator, &unparsed_indices);
+
+    if (RCL_RET_OK != ret) {
+      allocator.deallocate(unparsed_indices, allocator.state);
+      unparsed_indices = NULL;
+      return ret;
+    }
+
+    *nonros_argv = allocator.allocate(sizeof(char *) * *nonros_argc, allocator.state);
+    for (int i = 0; i < *nonros_argc; ++i) {
+      (*nonros_argv)[i] = argv[unparsed_indices[i]];
+    }
+
+    if (NULL != unparsed_indices) {
```

Comment:
This check might be redundant, since it's either non-free'd memory or null, both of which are fine to pass to deallocate. Also I think it must be non-null at this point, since you access it in the loop above so long as `nonros_argc > 0`.